docs/howto/custom-template-tags.txt

```diff
@@ -498,6 +498,195 @@ you see fit:
+
+.. admonition::  Content Escaping
+
+    ``simple_block_tag`` behaves similarly to ``simple_tag`` regarding
+    auto-escaping. For details on escaping and safety, refer to ``simple_tag``.
+    Because the ``content`` argument has already been rendered by Django, it is
+    already escaped.
+
+A more complete example
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Suppose a Django app, ``testapp``, needs to define a general message box to be
+used in templates, supporting various message levels and content beyond a
+simple phrase. The tag could be used as follows:
+
+.. code-block:: python
+    :caption: ``testapp/templatetags/testapptags.py``
+
+    from django import template
+    from django.utils.html import format_html
+
+
+    register = template.Library()
+
+
+    @register.simple_block_tag(takes_context=True)
+    def msgbox(context, content, level):
+        format_kwargs = {
+            "level": level.lower(),
+            "level_title": level.capitalize(),
+            "content": content,
+            "open": " open" if level.lower() == "error" else "",
+            "site": context.get("site", "My Site"),
+        }
+        result = """
+        <div class="msgbox {level}">
+          <details{open}>
+            <summary>
+              <strong>{level_title}</strong>: Please read for <i>{site}</i>
+            </summary>
+            <p>
+              {content}
+            </p>
+          </details>
+        </div>
+        """
+        return format_html(result, **format_kwargs)
+
+.. code-block:: python
+    :caption: ``testapp/views.py``
+
+    from django.shortcuts import render
+
+
+    def simpleblocktag_view(request):
+        return render(request, "test.html", context={"site": "Important Site"})
+
+
+Then, ``test.html`` would look like this:
+
+.. code-block:: html+django
+    :caption: ``testapp/templates/test.html``
+
+    {% extends "base.html" %}
+
+    {% load testapptags %}
+
+    {% block content %}
+
+      {% msgbox level="error" %}
+        Please fix all errors. Further documentation can be found at
+        <a href="http://example.com">Docs</a>.
+      {% endmsgbox %}
+
+      {% msgbox level="info" %}
+        More information at: <a href="http://othersite.com">Other Site</a>/
+      {% endmsgbox %}
+
+    {% endblock %}
+
+With a resulting HTML as follows:
+
+.. code-block:: html
+
+    <div class="msgbox error">
+      <details open>
+        <summary>
+          <strong>Error</strong>: Please read for <i>Important Site</i>
+        </summary>
+        <p>
+          Please fix all errors. Further documentation can be found at
+          <a href="http://example.com">Docs</a>.
+        </p>
+      </details>
+    </div>
+
+    <div class="msgbox info">
+      <details>
+        <summary>
+          <strong>Info</strong>: Please read for <i>Important Site</i>
+        </summary>
+        <p>
+          More information at: <a href="http://othersite.com">Other Site</a>
+        </p>
+      </details>
+    </div>
```

Comment:
I like that the example shows how this is useful in a more real way, I think it's too complete and I wouldn't put it in it's own section with a header. I would propose:
```suggestion
For a more complete example, a message box that supports different message
levels and content beyond a simple phrase could be implemented via a tag as
follows::

    from django import template
    from django.utils.html import format_html


    register = template.Library()


    @register.simple_block_tag()
    def msgbox(context, content, level, title):
        format_kwargs = {
            "level": level,
            "level_title": level.capitalize(),
            "content": content,
            "open": " open" if level == "error" else "",
            "title": title,
        }
        result = """
        <details{open} class="msgbox {level}">
          <summary>
            <strong>{level_title}!</strong>: {title}</i>
          </summary>
            <p>
              {content}
            </p>
          </details>
        """
        return format_html(result, **format_kwargs)


Then, the following template snippet:

.. code-block:: html+django

    {% msgbox level="error" title="You did something wrong!" %}
      Please fix all errors. Further documentation can be found at
      <a href="http://example.com">Docs</a>.
    {% endmsgbox %}

    {% msgbox level="info" title="Here is some help" %}
      More information at: <a href="http://othersite.com">Other Site</a>/
    {% endmsgbox %}

Would result in HTML as follows:

.. code-block:: html

    <details open class="msgbox error">
      <summary>
        <strong>Error!</strong>: You did something wrong!
      </summary>
      <p>
         Please fix all errors. Further documentation can be found at
         <a href="http://example.com">Docs</a>.
      </p>
    </details>

    <details open class="msgbox error">
      <summary>
        <strong>Info!</strong>: Here is some help
      </summary>
      <p>
        More information at: <a href="http://othersite.com">Other Site</a>
      </p>
    </details>

.. admonition::  Content Escaping

    ``simple_block_tag`` behaves similarly to ``simple_tag`` regarding
    auto-escaping. For details on escaping and safety, refer to ``simple_tag``.
    Because the ``content`` argument has already been rendered by Django, it is
    already escaped.
```

Reply:
Thank you @sarahboyce ! Following our conversation where we realized this is part of howto docs (with a mix of reference docs which prompted the initial comment), I pushed some wording tweaks to the example but otherwise left it as a more hand-holding description.